src/settings.plugin.js

```diff
@@ -321,26 +327,54 @@ wikiaJWPlayerSettingsPlugin.prototype.getSuitableCaptionsIndex = function(userLa
 		.indexOf(userLang);
 };
 
-wikiaJWPlayerSettingsPlugin.prototype.captionLangMap = {
-	en: 'English',
-	pl: 'Polski',
-	fr: 'Français',
-	de: 'Deutsch',
-	it: 'Italiano',
-	ja: '日本語',
-	pt: 'Português',
-	ru: 'Русский язык',
-	es: 'Español',
-	zh: '中文'
+wikiaJWPlayerSettingsPlugin.prototype.updateCurrentCaptions = function (data) {
+	for (var i = 0; i < this.captionsList.childNodes.length; i++) {
+		this.captionsList.childNodes[i].classList.remove(isActiveClass);
+	}
+
+	this.captionsList.querySelector('[data-track="' + data.track + '"]').classList.add(isActiveClass);
 };
 
-wikiaJWPlayerSettingsPlugin.prototype.getLabelElement = function (wrapper) {
-	return wrapper.querySelector('label');
+/**
+ * Hack warning!
+ *
+ * Programmatically set 'line' of every caption as a percentage value from the top of video
+ *
+ * @see https://www.w3.org/TR/webvtt1/#webvtt-cue-line
+ */
+wikiaJWPlayerSettingsPlugin.prototype.adjustCaptionsPosition = function () {
+	// timeout is used because of a problems on Safari, value is approximate
+	setTimeout(function () {
+		var textTracks = this.player.getContainer().querySelector('video').textTracks,
+			activeTrackIndex = Object.keys(textTracks).filter(function (key) {
+				/**
+				 * @see https://developer.mozilla.org/en-US/docs/Web/API/TextTrack/mode
+				 */
+				return textTracks[key].mode !== 'disabled';
+			})[0];
+
+		if (textTracks.length) {
+			Object.keys(textTracks[activeTrackIndex].cues).forEach(function (key) {
+				textTracks[activeTrackIndex].cues[key].snapToLines = false;
+				textTracks[activeTrackIndex].cues[key].line = 85;
+			});
+		}
+	}.bind(this), 50);
```

Comment:
I don't like timeouts with a magic number. Let's discuss it.

Can't we just listen to `captionsChanged` event?

Reply:
+1